src/Common/tests/System/Collections/ICollection.Generic.Tests.cs

```diff
@@ -0,0 +1,622 @@
+                });
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_Generic_Add_InvalidValueToBeginningOfCollection(int count)
+        {
+            if (!IsReadOnly)
+            {
+                Assert.All(InvalidValues, invalidValue =>
+                {
+                    ICollection<T> collection = GenericICollectionFactory(0);
+                    collection.Add(invalidValue);
+                    for (int i = 0; i < count; i++)
+                        collection.Add(TFactory(i));
+                    Assert.Equal(count, collection.Count);
+                });
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_Generic_Add_InvalidValueToEndOfCollection(int count)
+        {
+            if (!IsReadOnly)
+            {
+                Assert.All(InvalidValues, invalidValue =>
+                {
+                    ICollection<T> collection = GenericICollectionFactory(count);
+                    collection.Add(invalidValue);
+                    Assert.Equal(count, collection.Count);
+                });
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_Generic_Add_DuplicateValue(int count)
+        {
+            if (!IsReadOnly && DuplicateValuesAllowed)
+            {
+                ICollection<T> collection = GenericICollectionFactory(count);
+                T duplicateValue = TFactory(700);
+                collection.Add(duplicateValue);
+                collection.Add(duplicateValue);
+                Assert.Equal(count + 2, collection.Count);
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_Generic_Add_AfterCallingClear(int count)
+        {
+            if (!IsReadOnly)
+            {
+                ICollection<T> collection = GenericICollectionFactory(count);
+                collection.Clear();
+                AddToCollection(collection, 5);
+                Assert.Equal(5, collection.Count);
+            }
+        }
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_Generic_Add_AfterRemovingAnyValue(int count)
+        {
+            if (!IsReadOnly)
+            {
+                int seed = 840;
+                ICollection<T> collection = GenericICollectionFactory(count);
+                List<T> items = collection.ToList();
+                T toAdd = TFactory(seed++);
+                while (collection.Contains(toAdd))
+                    toAdd = TFactory(seed++);
+                collection.Add(toAdd);
+                collection.Remove(collection.ElementAt(0));
+
+                toAdd = TFactory(seed++);
+                while (collection.Contains(toAdd))
+                    toAdd = TFactory(seed++);
+
+                collection.Add(toAdd);
```

Comment:
Looks like I ported this test from the old ICollectionGenericTests.cs and forgot to add the part where compare ```collection``` to ```items```.